npm run all fails on get-all-submodules step

Node.JS v10.16.0
Ubuntu 18.04.2

Also tried to replace git with the https in .gitmodules, but script still trying to pull everything from git (magic?)

```
> posdao-test-setup@0.0.1 all /tmp/posdao-test-setup
> npm i && npm run get-all-submodules && npm run cleanup && npm run compile-posdao-contracts && npm run make-spec && npm run start-test-setup && npm run test && npm run stop-test-setup

audited 73948 packages in 3.359s
found 0 vulnerabilities


> posdao-test-setup@0.0.1 get-all-submodules /tmp/posdao-test-setup
> git submodule update --init --remote

Cloning into '/tmp/posdao-test-setup/posdao-contracts'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:poanetwork/posdao-contracts.git' into submodule path '/tmp/posdao-test-setup/posdao-contracts' failed
Failed to clone 'posdao-contracts'. Retry scheduled
Cloning into '/tmp/posdao-test-setup/posdao-contracts'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:poanetwork/posdao-contracts.git' into submodule path '/tmp/posdao-test-setup/posdao-contracts' failed
Failed to clone 'posdao-contracts' a second time, aborting
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! posdao-test-setup@0.0.1 get-all-submodules: `git submodule update --init --remote`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the posdao-test-setup@0.0.1 get-all-submodules script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/user/.npm/_logs/2019-06-24T06_42_17_113Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! posdao-test-setup@0.0.1 all: `npm i && npm run get-all-submodules && npm run cleanup && npm run compile-posdao-contracts && npm run make-spec && npm run start-test-setup && npm run test && npm run stop-test-setup`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the posdao-test-setup@0.0.1 all script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/user/.npm/_logs/2019-06-24T06_42_17_189Z-debug.log

```

--- a/.gitmodules
+++ b/.gitmodules
@@ -1,3 +1,3 @@
 [submodule "posdao-contracts"]
 	path = posdao-contracts
-	url = git@github.com:poanetwork/posdao-contracts.git
+  url = https://github.com/poanetwork/posdao-contracts.git

--- a/.gitmodules
+++ b/.gitmodules
@@ -1,3 +1,3 @@
 [submodule "posdao-contracts"]
 	path = posdao-contracts
-  url = https://github.com/poanetwork/posdao-contracts.git
+	url = https://github.com/poanetwork/posdao-contracts.git

--- a/package.json
+++ b/package.json
@@ -4,10 +4,10 @@
   "description": "",
   "scripts": {
     "all": "npm i && npm run get-all-submodules && npm run cleanup && npm run compile-posdao-contracts && npm run make-spec && npm run start-test-setup && npm run test && npm run stop-test-setup",
-    "cleanup": "rm -rf ./build ./parity-data ./accounts && git checkout ./parity-data && git checkout ./accounts",
+    "cleanup": "bash scripts/stop-test-setup && rm -rf ./build ./parity-data ./accounts && git checkout ./parity-data && git checkout ./accounts",
     "get-all-submodules": "git submodule update --init --remote",
     "compile-posdao-contracts": "cd ./posdao-contracts && npm i && npm run compile",
-    "make-spec": ". ./scripts/network-spec && cd ./posdao-contracts && npm i && node ./scripts/make_spec.js",
+    "make-spec": ". ./scripts/network-spec && cd ./posdao-contracts && npm i && node ./scripts/make_spec.js && node ../scripts/addCandidateBalances.js",
     "start-test-setup": "bash scripts/start-test-setup ../parity-ethereum/target/release/parity && (node scripts/watchOrdinaryNode.js &) && node scripts/deploy-staking-token.js",
     "test": "node_modules/.bin/mocha --bail --timeout 1200000 && npm run watcher",
     "stop-test-setup": "bash scripts/stop-test-setup",

--- a/scripts/deploy-staking-token.js
+++ b/scripts/deploy-staking-token.js
@@ -44,7 +44,6 @@ function compileContract() {
 async function main() {
     console.log('**** Check that StakingToken is already deployed in StakingAuRa');
     let existingStakingTokenAddress = await StakingAuRa.instance.methods.erc20TokenContract().call();
-    console.log('***** raw existingStakingTokenAddress = ', existingStakingTokenAddress);
     if (existingStakingTokenAddress
           && existingStakingTokenAddress.toLowerCase() != '0x'
           && existingStakingTokenAddress.toLowerCase() != '0x0000000000000000000000000000000000000000'

--- a/scripts/deploy-staking-token.js
+++ b/scripts/deploy-staking-token.js
@@ -42,6 +42,17 @@ function compileContract() {
 }
 
 async function main() {
+    console.log('**** Check that StakingToken is already deployed in StakingAuRa');
+    let existingStakingTokenAddress = await StakingAuRa.instance.methods.erc20TokenContract().call();
+    console.log('***** raw existingStakingTokenAddress = ', existingStakingTokenAddress);
+    if (existingStakingTokenAddress
+          && existingStakingTokenAddress.toLowerCase() != '0x'
+          && existingStakingTokenAddress.toLowerCase() != '0x0000000000000000000000000000000000000000'
+        ) {
+        console.log('***** StakingToken already deployed at ' + existingStakingTokenAddress + ', skipping deployment');
+        return;
+    }
+
     let compiledContract = compileContract();
     let abi = compiledContract.abi;
     let bytecode = compiledContract.evm.bytecode.object;

--- a/scripts/start-test-setup
+++ b/scripts/start-test-setup
@@ -6,6 +6,6 @@ set -e
 
 PARITY=$1
 for i in $(seq 0 6); do
-    "$PARITY" --config "./config/node${i}.toml" > "./parity-data/node${i}/log" 2>&1 &
+    "$PARITY" --config "./config/node${i}.toml" >> "./parity-data/node${i}/log" 2>&1 &
     node ./scripts/getReservedPeer.js "$i"
 done

--- a/scripts/start-test-setup
+++ b/scripts/start-test-setup
@@ -2,12 +2,10 @@
 
 # Exit on undefined variables.
 set -u
+set -e
 
 PARITY=$1
-NODE=`which node`
-$NODE ./scripts/addCandidateBalances.js
 for i in $(seq 0 6); do
-    rm -rf -- "./parity-data/node$i/chains"
     "$PARITY" --config "./config/node${i}.toml" > "./parity-data/node${i}/log" 2>&1 &
-    "$NODE" ./scripts/getReservedPeer.js "$i"
+    node ./scripts/getReservedPeer.js "$i"
 done

